chapters/annotations.tex

```diff
@@ -1657,19 +1657,31 @@ \subsubsection{Conversion rules}\doublelabel{conversion-rules}
 
 convertModifiers("Modelica.Blocks.Math.LinearDependency",
                  {"y0=0", "k1=0", "k2=0"}, {"y0=%y0%", "k1=%y0%*%k1%", "k2=%y0%*%k2%"}, true)
+convertClass("Modelica.Electrical.Machines.BasicMachines.QuasiStationaryDCMachines",
+              "Modelica.Electrical.Machines.BasicMachines.QuasiStaticDCMachines")
+convertElement({"Modelica.Electrical.Machines.Interfaces.PartialBasicDCMachine"},
+                "quasiStationary", "quasiStatic")
+convertElement("Modelica.Electrical.Machines.BasicMachines.QuasiStationaryDCMachines.DC_ElectricalExcited",
+                "quasiStationary", "quasiStatic")
 \end{lstlisting}
 converts
 \begin{lstlisting}[language=modelica]
 Modelica.Thermal.FluidHeatFlow.Components.IsolatedPipe pipe1;
 Modelica.StateGraph.Temporary.NumericValue tempValue(Value=10, hideConnector=true);
 Modelica.Blocks.Math.LinearDependency linearDep(y0=2, k2=1);
+Modelica.Electrical.Machines.BasicMachines.QuasiStationaryDCMachines.DC_ElectricalExcited a;
+Boolean b=a.quasiStationary;
 \end{lstlisting}
 to
 \begin{lstlisting}[language=modelica]
 Modelica.Thermal.FluidHeatFlow.Components.Pipe pipe1(useHeatPort=false);
 Modelica.Blocks.Interaction.Show.RealValue(number=10, use_numberPort=not true);
 Modelica.Blocks.Math.LinearDependency linearDep(y0=2, k1=0, k2=2);
+Modelica.Electrical.Machines.BasicMachines.QuasiStaticDCMachines.DC_ElectricalExcited a;
+Boolean b=a.quasiStatic;
 \end{lstlisting}
+The \lstinline!convertElement! call for \lstinline!DC_ElectricalExcited! is needed to avoid relying on base-classes
+in the original library where \lstinline!DC_ElectricalExcited! inherit from \lstinline!PartialBasicDCMachine!.
```

Comment:
It looks like there is a typo:  "inherit" should be "inherits"

I find this example hard to understand. I had to open the MSL to look up the definitions and dependencies.  Is the following a correct interpretation:

In the original library, `DC_ElectricalExcited` extends from `PartialBasicDCMachine` and inherits from it an element called `quasiStationary`. In the new version of the library, `quasiStationary` was renamed to `quasiStatic` inside `PartialBasicDCMachine`. So we need to define convertElement for `quasiStationary` in `PartialBasicDCMachine` and in any class that extends from it.

Is it right? What if there is a class in a user library that like `DC__ElectricalExcited` extends from `PartialBasicDCMachine` and makes some references to `quasiStationary`?

By the way, `quasiStationary` is declared as protected in `PartiaBasicDCMachine` in MSL 3.2.3. Shouldn't it be illegal to access it via the dot notation as is done on line 1673?

Reply:
Starting from the bottom: Yes, it is illegal to access it that way. I believe it would be possible to avoid that - and also clarify the point above.

Reply:
The following should now be corrected:

- Spelling
- Illegal use of protected element
- Added example of inheritance among user-models